File: c1/p2/ans.cpp
#include <bits/stdc++.h>
using namespace std;

const string names[] = {"miso", "hakuto", "ringo", "paula", "sprite"}; 

string s;

signed main() {
    ios_base::sync_with_stdio(0);cin.tie(0);

    cin >> s;
    cout << (s == "miso" || s == "hakuto" || s == "ringo" || s == "paula" || s == "sprite" ? "no" : "yes");

    return 0;
}

File: c1/p2/ans.py
# print("no" if input() in ["miso", "hakuto", "ringo", "paula", "sprite"] else "yes")

names = ["miso", "hakuto", "ringo", "paula", "sprite"]
gpt_name = input()
if gpt_name in names:
    print("no")
else:
    print("yes")